GG/src/GLClientAndServerBuffer.cpp

```diff
@@ -170,7 +170,7 @@ void GL2DVertexBuffer::activate() const
         glBindBuffer(GL_ARRAY_BUFFER, b_name);
         glVertexPointer(2, GL_FLOAT, 0, nullptr);
         glBindBuffer(GL_ARRAY_BUFFER, 0);
-    } else {
+    } else if (!b_data.empty()) {
```

Comment:
Could this leave OpenGL in an unexpected state depending on what was preivously bound?

Reply:
Can't speak to OpenGL state, the previous code could modify `b_data` and pass an uninitialized value.

Reply:
where is/was `b_data` being modified? `operator[]` doesn't add elements to a vector.

It's passing a pointer to where the data array of `b_data` would be if there were some. Doesn't necessarily mean there's a problem, since the `count` parameter to the subsequent `glDrawArrays(...)` call should be `0` if the appropriate vector size is passed and that vector is empty.

Reply:
Somehow I forgot about the differing behavior of `operator[]` :/
I've minimized the changes to ListBox.

Might look at defining a refactoring cells and cell properties into a new Row::Cell struct after 0.4.8 is finalized.